t/90h2olog.t

```diff
@@ -0,0 +1,91 @@
+#!perl
+# DTRACE_TESTS=1 to skip to check prereqisites
+# H2OLOG_DEBUG=1 for more runtime logs
+use strict;
+use warnings FATAL => "all";
+use Net::EmptyPort qw(empty_port);
+use Test::More;
+use JSON;
+use t::Util;
+
+run_as_root();
+
+my $h2olog_prog = bindir() . "/h2olog";
+my $client_prog = bindir() . "/h2o-httpclient";
+
+unless ($ENV{DTRACE_TESTS})  {
+  plan skip_all => "$h2olog_prog not found"
+      unless -e $h2olog_prog;
+
+  plan skip_all => "$client_prog not found"
+      unless -e $client_prog;
+
+  plan skip_all => 'dtrace support is off'
+      unless server_features()->{dtrace};
+}
+
+my $quic_port = empty_port({
+    host  => "127.0.0.1",
+    proto => "udp",
+});
+
+my $server = spawn_h2o({
+    opts => [qw(--mode=worker)],
+    conf => << "EOT",
+listen:
+  type: quic
+  port: $quic_port
+  ssl:
+    key-file: examples/h2o/server.key
+    certificate-file: examples/h2o/server.crt
+hosts:
+  default:
+    paths:
+      /:
+        file.dir: t/assets/doc_root
+EOT
+});
+
+subtest "h2olog", sub {
+  my $tracer = H2ologTracer->new({
+    pid => $server->{pid},
+    args => [],
+  });
+
+  my ($headers, $body) = run_prog("$client_prog -3 https://127.0.0.1:$quic_port/");
+  like $headers, qr{^HTTP/3 200\n}, "req: HTTP/3";
+
+  my $trace;
+  until (($trace = $tracer->get_trace()) =~ m{"h3s-destroy"}) {}
+
+  if ($ENV{H2OLOG_DEBUG}) {
+    diag "h2olog output:\n", $trace;
+  }
+
+  ok( (map { decode_json($_) } split /\n/, $trace), "h2olog output is valid JSON Lines");
+};
+
+subtest "h2olog -H", sub {
+  my $tracer = H2ologTracer->new({
+    pid => $server->{pid},
+    args => ["-H"],
+  });
+
+  my ($headers, $body) = run_prog("$client_prog -3 https://127.0.0.1:$quic_port/");
+  like $headers, qr{^HTTP/3 200\n}, "req: HTTP/3";
+
+  my $trace;
+  until (($trace = $tracer->get_trace()) =~ m{RxProtocol\s+HTTP/3.0}) {}
+
+  if ($ENV{H2OLOG_DEBUG}) {
+    diag "h2olog output:\n", $trace;
+  }
+
+  ok length($trace), "h2olog output exists"
```

Comment:
I wonder what we are trying to test here... Isn't this condition satisfied by the `until` loop above, that tests that `$trace` contains a certain string?

Reply:
Oh, that's a good suggestion. It tests nothing. Let's test something useful. 0c61af816624bd80c05777ee58d61534853de9d4